ReactiveCocoa/Swift/Property.swift

```diff
@@ -134,50 +132,21 @@ FIXME: DISABLED DUE TO COMPILER BUG
 /// The binding will automatically terminate when the property is deinitialized,
 /// or when the created signal sends a `Completed` event.
 public func <~ <T>(property: MutableProperty<T>, producer: SignalProducer<T, NoError>) -> Disposable {
-	let disposable = CompositeDisposable()
-	let propertyDisposable = property.producer.start(completed: {
-		disposable.dispose()
-	})
-
-	disposable.addDisposable(propertyDisposable)
-
+	var disposable: Disposable!
+	
 	producer.startWithSignal { signal, signalDisposable in
-		disposable.addDisposable(signalDisposable)
-
-		signal.observe(next: { [weak property] value in
-			property?.value = value
-			return
-		}, completed: {
-			disposable.dispose()
-		})
+		property <~ signal
```

Comment:
But if the property deinitializes, the inner <~ will take care of disposing the Signal. Shouldn't that be enough?